src/overlays/actors/ovl_Obj_Syokudai/z_obj_syokudai.c

```diff
@@ -27,11 +29,243 @@ const ActorInit Obj_Syokudai_InitVars = {
+                    } else {
+                        this->timer = (litTimeScale * 50) + 110;
+                    }
+                }
+                Audio_PlaySoundGeneral(NA_SE_EV_FLAME_IGNITION, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            }
+        }
+    }
+    collider1 = &this->collider1;
+    Collider_CylinderUpdate(&this->actor, collider1);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &collider1->base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &collider1->base);
+    
+    collider2 = &this->collider2;
+    Collider_CylinderUpdate(&this->actor, collider2);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &collider2->base);
+
+    if (this->timer > 0) {
+        this->timer--;
+        if ((this->timer == 0) && (torchType != 0)) {
+            sLitTorchCount--;
+        }
+    }
+    if (this->timer != 0) {
+        if ((this->timer < 0) || (this->timer >= 20)) {
+            lightRadius = 200;
+        } else {
+            lightRadius = (this->timer * 200.0f) / 20.0f;
+        }
+        lightColorY = (u8)(Math_Rand_ZeroOne() * 127.0f) + 128;
+        func_8002F974(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    Lights_PointSetColorAndRadius(&this->lightInfo, lightColorY, lightColorY, 0, lightRadius);
+    this->flameTexScroll++;
+}
+
+void ObjSyokudai_Draw(Actor* thisx, GlobalContext *globalCtx) {
+    static Gfx* sDisplayLists[] = {0x060003A0, 0x06000B90, 0x06000870};
+    s32 pad;
+    ObjSyokudai* this = THIS;
+    s32 timerMax;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Syokudai/ObjSyokudai_Update.s")
+    timerMax = (((this->actor.params >> 6) & 0xF) * 50) + 100;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Syokudai/ObjSyokudai_Draw.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_obj_syokudai.c", 707);
+    func_80093D18(globalCtx->state.gfxCtx);
+    
+    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_obj_syokudai.c", 714), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    
+    gSPDisplayList(oGfxCtx->polyOpa.p++, sDisplayLists[(u16)this->actor.params >> 0xC]);
+    
+    if (this->timer != 0) {
+        f32 flameScale = 1.0f;
+        
+        if (this->timer > timerMax) {
+            flameScale = (timerMax - this->timer + 10) / 10.0f;
+        } else if ((this->timer > 0) && (this->timer < 20)) {
+            flameScale = this->timer / 20.0f;
+        }
+        flameScale *= 0.0027f;
+        
+        func_80093D84(globalCtx->state.gfxCtx);
+        
+        gSPSegment(oGfxCtx->polyXlu.p++, 0x08, Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0, (this->flameTexScroll * -20) & 0x1FF, 0x20, 0x80));
+        
+        gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0x80, 0x80, 255, 255, 0, 255);
+        
+        gDPSetEnvColor(oGfxCtx->polyXlu.p++, 255, 0, 0, 0);
+        
+        Matrix_Translate(0.0f, 52.0f, 0.0f, 1);
```

Comment:
```suggestion
        Matrix_Translate(0.0f, 52.0f, 0.0f, MTXMODE_APPLY);
```
and other `Matrix_` calls too